### ISSUE TITLE
Can't connect to socket without supervisorctl stop and start

### ISSUE THREAD

**Issue by ProbablyDom** [bbernhard/signal-cli-rest-api#378]
### The problem

TL;DR: `receiveData` won't receive any data unless I log into the running docker container, stop the running CLI and start it again.

I started using the REST API to integrate with my own tooling.  Just to make sure I did not do anything wrong I used your golang client to connect. Sending works without issues. But I struggled a lot with receiving. I logged into the container and stopped the cli. Then started the cli directly and got messages on standard out. I then started the cli with the same command the entrypoint script is using and was able to receive messages via the golang client.

After some testing I realized I can just log into the container, do a `supervisorctl stop all`, kill the Java process, and then `supervisorctl start all`. Then receiving starts until I restart the container.

Before I automate those steps I wanted to check if that is a known issue and if there is an existing solution available.

I also tried to read the socket inside the docker container via `nc localhost 6000` but that is also not successful without the mentioned procedure.

OS: `Ubuntu 22.04.2 LTS`

`$ docker version`
```
Client:
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.1
 Git commit:        20.10.21-0ubuntu1~22.04.3
 Built:             Thu Apr 27 05:57:17 2023
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
```

`jsonrpc2.yml`
```
config:
  "+1234567890":
    tcp_port: 6000
    fifo_pathname: /tmp/sigsocket0
```

`docker-compose.yaml`
```
version: "3"
services:
  signal-cli-rest-api:
    container_name: signal-api
    restart: unless-stopped
    image: bbernhard/signal-cli-rest-api:latest
    environment:
      - MODE=json-rpc
    ports:
      - "2555:8080"
      - "6000:6000"
    volumes:
      - "/home/my-user/signal-cli-rest-api/config:/home/.local/share/signal-cli"
  ```

### Are you using the latest released version?

- [X] Yes

### Have you read the troubleshooting page?

- [X] Yes

### What type of installation are you running?

signal-cli-rest-api Docker Container

### In which mode are you using the docker container?

JSON-RPC Mode

### What's the architecture of your host system?

x86-64

**Comment by ProbablyDom** [comment]
Small update. I realized that after stopping and starting the REST API doesn't work anymore. Sending messages to it will result in:

```
{"error":"write tcp 127.0.0.1:41666-\u003e127.0.0.1:6000: write: broken pipe"}
```

So essentially I can either send or receive, but not both :(

Any ideas?

**Comment by bbernhard** [comment]
Thanks for the detailed bug report! That's really strange. :thinking: 

If you switch to a different mode (e.g `native`) does it work then? Was the `json-rpc` mode working at some point and broke with an update or was it never working?

**Comment by ProbablyDom** [comment]
I have used `normal` at the beginning and it works fine, except for the high latency of course. That's why I switched to `json-rpc`.

Just checked again, switched to `normal`, receiving and sending works. Switched back to `json-rpc` and still the same issue.


I just started to use your image a few days ago. Never used any other version. I could try to downgrade to an older version if you want me to test that. What would be a good version to use for that?

**Comment by ProbablyDom** [comment]
I didn't see anything in the logs. So I switched to the debug logs. I realized a lot of things are actually logged in `debug`.

And I get this every time a message is received:

```
time="2023-06-29T00:04:26Z" level=debug msg="Couldn't send message to golang channel, as there's no receiver"
```
